actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -285,7 +285,7 @@ def write_fragment_for(name, options)
         pos = output_buffer.length
         yield
         output_safe = output_buffer.html_safe?
-        fragment = output_buffer.slice!(pos..-1)
+        fragment = output_buffer.to_s.slice!(pos..-1)
```

Comment:
I don't think this change is quite right. Previously this was actually mutating the output buffer (it's a really strange thing to do). @seejohnrun and I ran into this as well when experimenting with modifying outputbuffer, and tried replacing what this method does `capture` https://github.com/jhawthorn/rails/pull/1/files#diff-7e8f6bc4873643df271aa007b513b993a8e424964b76e4eec2c1d156d9108ac3R252 (but I don't remember if it works)

Reply:
I'll look more into it tomorrow, but I was under the impression that mutating the buffer was simply an unneeded side effect. I'm pretty sure that buffer instance is no longer used after that `slice!`.

Reply:
Ok so I was wrong. This `slice!` is to handle formats such as `.builder`, as it doesn't access `@output_buffer`, it is initialized once with it.

So in `cache do` we can't just substitute the buffer, we have to mutate it.

That's really unfortunate.

I'm tempted to add some stack to the buffer, the problem is that it's not clear to be wether `@output_buffer` is guaranteed to be an `OutputBuffer` (or `StreamBuffer`) as there are some tests using `ActiveSupport::SafeBuffer`. I'd have to dig into that as well to see wether it's legit or not.